src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_collect.c

```diff
@@ -1,103 +1,1003 @@
+                // shadow
+                gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, pauseCtx->alpha);
+            } else {
+                if (Inventory_GetSkullTokenCount(play->sceneId) == 100) {
+                    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 200, 50, 50, pauseCtx->alpha);
+                } else {
+                    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+                }
+            }
+
+            isDigitDrawn = false;
+            for (j = 0; j < 3; j++, i += 4) {
+                if ((j >= 2) || (skullTokenDigits[j] != 0) || isDigitDrawn) {
+                    gDPLoadTextureBlock(POLY_OPA_DISP++, ((u8*)gCounterDigit0Tex + (8 * 16 * skullTokenDigits[j])),
+                                        G_IM_FMT_I, G_IM_SIZ_8b, 8, 16, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                                        G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+                    gSP1Quadrangle(POLY_OPA_DISP++, i, i + 2, i + 3, i + 1, 0);
+
+                    isDigitDrawn = true;
+                }
+            }
+        }
+    }
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+#else
+s16 sQuestRemainsColorTimer = 20;
+s16 sQuestRemainsColorTimerIndex = 0;
+s16 sQuestHpPrimRed = 0;
+s16 sQuestHpPrimGreen = 0;
+s16 sQuestHpPrimBlue = 0;
+s16 sQuestHpPrimAlpha = 0;
+s16 sQuestHpColorTimer = 20;
+s16 sQuestHpPrimColorTargetIndex = 0;
+TexturePtr sOcarinaButtonTextures[] = {
+    gOcarinaATex, gOcarinaCDownTex, gOcarinaCRightTex, gOcarinaCLeftTex, gOcarinaCUpTex,
+};
+s16 sQuestSongsPrimRed[] = {
     150, 255, 100, 255, 255, 255, 255, 255, 255, 255, 255, 255,
 };
-
-s16 D_8082AFDC[] = {
+s16 sQuestSongsPrimGreen[] = {
     255, 80, 150, 160, 100, 240, 255, 255, 255, 255, 255, 255,
 };
+s16 sQuestSongsPrimBlue[] = {
+    100, 40, 255, 0, 255, 100, 255, 255, 255, 255, 255, 255,
+};
+TexturePtr sQuestUpgradeTextures[][3] = {
+    { 0x08053000, 0x08054000, 0x08055000 },
+    { 0x08056000, 0x08057000, 0x08058000 },
+};
+u8 sQuestUpgrades[] = { UPG_QUIVER, UPG_BOMB_BAG };
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/KaleidoScope_DrawQuestStatus.s")
+#endif
 
-s16 D_8082AFF4[] = { 100, 40, 255, 0, 255, 100, 255, 255, 255, 255, 255, 255 };
+typedef enum {
+    /* -3 */ CURSOR_TO_LEFT = -3, // Cursor on the "scroll to left page" position
+    /* -2 */ CURSOR_TO_RIGHT,     // Cursor on the "scroll to right page" position
+    /* -1 */ CURSOR_NONE          // No position in that direction, cursor stays where it is
+} CursorPointNext;
 
-void* D_8082B00C[] = {
-    0x08053000, 0x08054000, 0x08055000, 0x08056000, 0x08057000, 0x08058000,
-};
+typedef struct {
+    /* 0x0 */ s16 up;
+    /* 0x2 */ s16 down;
+    /* 0x4 */ s16 left;
+    /* 0x6 */ s16 right;
+} CursorPointDirection;
 
-s16 D_8082B024 = 1;
+void KaleidoScope_UpdateQuestCursor(PlayState* play) {
+    static s16 sQuestSongPlaybackDelayTimer = 0;
+    static CursorPointDirection sCursorPointLinks[] = {
```

Comment:
Here too?

Reply:
Yeah, this one is very unfortunate. `sQuestSongPlaybackDelayTimer` specifically affects regalloc, and `sCursorPointLinks` does to maintain ordering

Reply:
Well I *suppose* you could forward-declare it, but no thanks.